aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/CreateHandler.java

```diff
@@ -210,10 +216,29 @@ private String fetchEngine(final ProxyClient<RdsClient> client, final ResourceMo
         }
 
         if (ResourceModelHelper.isDBInstanceReadReplica(model)) {
-            return fetchDBInstance(client, model.getSourceDBInstanceIdentifier()).engine();
+            final String sourceDBInstanceArn = model.getSourceDBInstanceIdentifier();
+            final String sourceDBInstance = ResourceModelHelper.isValidArn(sourceDBInstanceArn) ?
+                    ResourceModelHelper.getResourceNameFromArn(sourceDBInstanceArn) : sourceDBInstanceArn;
+            if (ResourceModelHelper.isCrossRegionDBInstanceReadReplica(model, currentRegion)) {
+                final String sourceRegion = ResourceModelHelper.getRegionFromArn(sourceDBInstanceArn);
+                final ProxyClient<RdsClient> sourceRegionClient = new LoggingProxyClient<>(logger,
```

Comment:
Are we instantiating the client multiple times? For both cases we are using the source region for the client.

Reply:
If It is cross-region read replica dbinstance, the code will return the fetched engine without instantiating the client again. Therefore, it will never be instantiated again in DBCluster case.